### PR TITLE
fix(observability docs): restructure docs for v2 telemetry (and remove problematic Mixer refs)

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR is a proposal for removing Mixer-specific content from a couple of key places, while preserving the content where needed. This is part of the work involved in making the telemetry v2 work on-by-default in 1.5.

In particular, the concept guides for Observability and "What is Istio?" have mentions of Mixer removed. The Mixer-specific Tasks are moved into a sub-category of Observability Tasks, as Mixer is merely deprecated but not removed. The 1.3 Alpha Task on enabling v2 telemetry is also removed, as it is no longer relevant. Finally, some of the old links are updated, along with Mixer-focused bits of the microservices example.

Notably, new content is not added in this PR. This includes any work to "fix" the existing Mixer-specific Tasks by including steps to enable Mixer, as well as any new content to discuss the v2 architecture.

Also not addressed are any updates to the "Ops" guides on troubleshooting metrics and on deployment architecture.

Signed-off-by: Douglas Reid <dougreid@google.com>

### CODE REVIEW

File: content/en/docs/tasks/observability/mixer/_index.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: Using Mixer for Telemetry
```
Comment (frankbu) [inline content/en/docs/tasks/observability/mixer/_index.md]: Shouldn't this and all the nested docs be explicitly marked as deprecated?
```suggestion
title: Using Mixer for Telemetry (deprecated)
```

**Comment by douglas-reid** [comment]
Gentle ping for docs reviewers. I would like to start editing the content further to provide v2 / v1 updates, but want to ensure that I'm headed in the right direction first.

**Comment by frankbu** [comment]
@douglas-reid LGTM. Do you want to merge this one and do the following work in a different PR, or do you want to keep this one a WIP? If the latter, you should add the wip label.

**Comment by douglas-reid** [comment]
I'd like to do the work in a set of small follow-on PRs.

**Pull request merged**